test/custom_operator/op.cpp

```diff
@@ -0,0 +1,37 @@
+#include <torch/op.h>
+
+#if defined(WITH_PYTHON_OP_BINDINGS)
+#include <pybind11/pybind11.h>
+#endif
+
+#include <cstddef>
+#include <vector>
+
+std::vector<at::Tensor> custom_op(
+    at::Tensor tensor,
+    double scalar,
+    int64_t repeat) {
+  std::vector<at::Tensor> output;
+  output.reserve(repeat);
+  for (int64_t i = 0; i < repeat; ++i) {
+    output.push_back(tensor * scalar);
+  }
+  return output;
+}
+
+torch::RegisterOperators registry("custom::op", &custom_op);
+
+#if defined(WITH_PYTHON_OP_BINDINGS)
```

Comment:
you don't need to create python module. If you create the shared library that works through the registration mechanism, it's sufficient to just load it into memory and it can be done ctypes.CDLL

actually you can verbatim copy the mechanism Caffe2 uses as it's exactly the same problem. See:
* https://github.com/pytorch/pytorch/blob/HEAD/caffe2/python/dyndep.py
* https://github.com/pytorch/pytorch/blob/HEAD/caffe2/python/extension_loader.py (handles RTLD_GLOBAL)
* pytorch already handles RTLD_GLOBAL for the main _C.so: https://github.com/pytorch/pytorch/blob/HEAD/torch/__init__.py#L75

The only benefit of setup.py I can see is that setuptools manages the install path for the package. However, it depends on what's the target workflow you imagine for people developing c++ extensions. Would they install it to system paths? Or would they just drop the .so in the current model folder and load it directly?

Reply:
Thanks for the cdll links, looks awesome! For C++ extensions setuptools helped us getting the paths to Python.h.

I think because these JIT/script extensions are intended more for production use cases, compared to non-JIT C++ extensions, the pure .so path without setuptools might be the one to advertise